vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/strategies/MobileAppFullShootingStrategy.java

```diff
@@ -0,0 +1,330 @@
+import org.vividus.mobileapp.configuration.MobileApplicationConfiguration;
+import org.vividus.mobileapp.model.SwipeDirection;
+import org.vividus.reporter.event.AttachmentPublisher;
+import org.vividus.selenium.mobileapp.MobileAppWebDriverManager;
+import org.vividus.selenium.screenshot.ScreenshotUtils;
+import org.vividus.selenium.screenshot.strategies.ScreenshotShootingStrategy;
+import org.vividus.ui.util.ImageUtils;
+import org.vividus.util.Sleeper;
+
+import boofcv.alg.template.TemplateMatching;
+import boofcv.factory.template.FactoryTemplateMatching;
+import boofcv.factory.template.TemplateScoreType;
+import boofcv.io.image.ConvertBufferedImage;
+import boofcv.struct.feature.Match;
+import boofcv.struct.image.GrayF32;
+import ru.yandex.qatools.ashot.coordinates.Coords;
+import ru.yandex.qatools.ashot.shooting.ShootingStrategy;
+import ru.yandex.qatools.ashot.util.ImageTool;
+
+@Named("FULL")
+public class MobileAppFullShootingStrategy implements ShootingStrategy, ScreenshotShootingStrategy
+{
+    private static final long serialVersionUID = -1367316946809918634L;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MobileAppFullShootingStrategy.class);
+
+    private static final Duration SWIPE_DURATION = Duration.ofSeconds(2);
+    private static final int DEFAULT_SHIFT = 100;
+    private static final int DIVIDER = 3;
+    private static final double CUT_PERCENTAGE = 0.15;
+    private static final String UNSUPPORTED_MESSAGE = "Element screenshot is not supported";
+    private static final double MATCH_SCORE_THRESHOLD = 0.99d;
+
+    private final transient TouchActions touchActions;
+    private final transient MobileAppWebDriverManager genericWebDriverManager;
+    private final transient MobileApplicationConfiguration mobileApplicationConfiguration;
+    private final transient AttachmentPublisher attachmentPublisher;
+
+    public MobileAppFullShootingStrategy(TouchActions touchActions,
+            MobileAppWebDriverManager genericWebDriverManager,
+            MobileApplicationConfiguration mobileApplicationConfiguration, AttachmentPublisher attachmentPublisher)
+    {
+        this.touchActions = touchActions;
+        this.genericWebDriverManager = genericWebDriverManager;
+        this.mobileApplicationConfiguration = mobileApplicationConfiguration;
+        this.attachmentPublisher = attachmentPublisher;
+    }
+
+    @Override
+    public ShootingStrategy getDecoratedShootingStrategy(ShootingStrategy shootingStrategy)
+    {
+        return this;
+    }
+
+    @Override
+    public BufferedImage getScreenshot(WebDriver webDriver)
+    {
+        try
+        {
+            touchActions.swipeUntilEdge(SwipeDirection.DOWN, SWIPE_DURATION);
+            int height = genericWebDriverManager.getSize().getHeight();
+            int cutSize = (int) (height * CUT_PERCENTAGE);
+            return getScreenshot(webDriver, ImageArea.from(cutSize, height - cutSize * 2), DEFAULT_SHIFT);
```

Comment:
make CUT_PERCENTAGE = 30 instead?

Reply:
cutSize plays role in top offset and scrollable area computations so can't just make 30